### PR TITLE
feat(tenable): add Tenable assets and vulnerability findings intel module

### LINKED ISSUE #2269: feat(Tenable): Add support for Tenable export API
# Add a new intel Module to use Tenable Export API to gather assets and scans data.

## Description
Multiple companies use Tenable as their vulnerability management platform. The raw vulnerabilities data 

## Motivation
We use Tenable to scan network assets and would like to ingest the data from the scans and assets included.
The native API to export sucha the data in batches are the export API's, where a export request is sent and a subsequent export downloads is triggered based previous request.

## Alternatives Considered
Using other Tenable API's, but they are limited to less number of results.

## Relevant Links
- https://developer.tenable.com/reference/export-assets-v2
- https://developer.tenable.com/reference/exports-assets-download-chunk
- https://developer.tenable.com/reference/exports-vulns-request-export
- https://developer.tenable.com/reference/exports-vulns-download-chunk
- https://developer.tenable.com/docs/concurrency-limiting
- https://developer.tenable.com/docs/rate-limiting

### PR DESCRIPTION
**Pull request by heryxpc**
### Type of change
- [x] New feature (non-breaking change that adds functionality)
- [x] Documentation update

### Summary

Adds a new Tenable intel module that syncs assets and vulnerability findings from the [Tenable Export API](https://developer.tenable.com/reference/export-assets-v2).

**Assets sync** (`cartography/intel/tenable/assets.py`):
- Exports all assets via the async bulk-export workflow (POST → poll → download chunks)
- Produces `TenableAsset` nodes plus dedicated sub-nodes for cloud provider details (`TenableAssetAWS`, `TenableAssetAzure`, `TenableAssetGCP`), logical networks (`TenableNetwork`), scan sources (`TenableAssetSource`), and key/value tags (`TenableAssetTag`)

**Findings sync** (`cartography/intel/tenable/findings.py`):
- Exports vulnerability findings with a configurable lookback window (default 180 days) using the `last_found` filter so every sync is a full re-fetch of the window
- Produces `TenableFinding` nodes (labelled `CVE` when a CVE ID is present for ontology matching), `TenablePlugin` nodes (deduplicated across findings, carrying CVSS/VPR/EPSS scores and CVE lists), and `TenableScan` nodes
- Cleanup jobs prune stale nodes outside the window on every run

**New CLI flags** (all optional):
- `--tenable-access-key-env-var` / `--tenable-secret-key-env-var` — API credentials
- `--tenable-url` — base URL (default `https://cloud.tenable.com`)
- `--tenable-tenant-id` — container UUID from Tenable account settings; defaults to URL hostname
- `--tenable-findings-lookback-days` — lookback window in days (default `180`)

### Related issues or links

- Fixes https://github.com/cartography-cncf/cartography/issues/2269
- Tenable asset export API docs: https://developer.tenable.com/docs/retrieve-asset-data-from-tenableio
- Tenable vulnerability export API docs: https://developer.tenable.com/docs/retrieve-vulnerability-data-from-tenableio

### How was this tested?

- **Unit tests** (`tests/unit/cartography/intel/tenable/`): 46 tests covering all transform functions for assets and findings — field mapping, deduplication, edge cases (missing fields, empty input, null values)
- **Integration tests** (`tests/integration/cartography/intel/tenable/`): end-to-end sync tests exercising `sync()` against a live Neo4j testcontainer, asserting correct node properties, relationships, stale-node cleanup for every node type (`TenableAsset`, `TenableNetwork`, `TenableAssetSource`, `TenableAssetTag`, `TenableAssetAWS`, `TenableAssetAzure`, `TenableAssetGCP`, `TenableFinding`, `TenablePlugin`, `TenableScan`)

#### General
- [x] I have read the [contributing guidelines](https://cartography-cncf.github.io/cartography/dev/developer-guide.html).
- [x] The linter passes locally (`make test_lint`).
- [x] I have added/updated tests that prove my fix is effective or my feature works.

#### Proof of functionality
- [x] New or updated unit/integration tests.

#### If you are adding or modifying a synced entity
- [ ] Included Cartography sync logs from a real environment demonstrating successful synchronization of the new/modified entity.

#### If you are changing a node or relationship
- [x] Updated the [schema documentation](https://github.com/cartography-cncf/cartography/tree/master/docs/root/modules).

#### If you are implementing a new intel module
- [x] Used the NodeSchema [data model](https://cartography-cncf.github.io/cartography/dev/writing-intel-modules.html#defining-a-node).

### Notes for reviewers

The findings sync deliberately does a **full re-fetch every run** (no incremental cursor) so the `GraphJob` cleanup can correctly prune findings that have fallen outside the lookback window. The `--tenable-tenant-id` flag should be set to the container UUID shown in Tenable account settings when running against a non-default endpoint or managing multiple tenants, so graph nodes are scoped correctly.

### CODE REVIEW

File: cartography/cli.py [hunk 1]
```diff
@@ -1711,6 +1713,61 @@ def run(
...
+            tenable_secret_key_env_var: Annotated[
+                str,
+                typer.Option(
+                    "--tenable-secret-key-env-var",
+                    help="Environment variable name containing the Tenable secret key.",
+                    rich_help_panel=PANEL_TENABLE,
+                    hidden=PANEL_TENABLE not in visible_panels,
+                ),
+            ] = "TENABLE_SECRET_KEY",
+            tenable_findings_lookback_days: Annotated[
```
Comment (jychp) [inline cartography/cli.py]: `--tenable-findings-lookback-days` accepts negative values. A negative value makes `since_epoch = update_tag - days*86400` land in the future, so the export returns nothing and the cleanup job then deletes every finding in the graph. Enforce a non-negative bound with `typer.Option(..., min=0)`.

File: cartography/intel/tenable/api.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+def _initiate_export(
+    session: requests.Session,
+    base_url: str,
+    export_path: str,
+    params: dict[str, Any],
+) -> str:
+    """POST to start an async export job. Returns the export UUID."""
+    url = f"{base_url}/{export_path}"
+    response = session.post(url, json=params, timeout=(30, 30))
```
Comment (jychp) [inline cartography/intel/tenable/api.py]: `_initiate_export` does not handle Tenable's 409 active-export response. When a prior export is still running (overlapping schedule, or a retry after Cartography timed out), Tenable returns 409 with the in-flight export id instead of starting a new one; `raise_for_status()` aborts the whole sync. Catch the 409 and adopt the active export id.

File: tests/integration/cartography/intel/tenable/test_findings.py [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+        "TenableFinding",
+        "id",
+        "RESOURCE",
+        rel_direction_right=True,
+    )
+    assert {
+        (TENABLE_TENANT_ID, FINDING_ID_1),
+        (TENABLE_TENANT_ID, FINDING_ID_2),
+        (TENABLE_TENANT_ID, FINDING_ID_3),
+    } <= actual_rels
```
Comment (jychp) [inline tests/integration/cartography/intel/tenable/test_findings.py]: Subset assertion (`expected <= actual_rels`) lets a missing tenant->finding edge pass undetected. Use exact equality (`==`) like the other relationship tests in this file.

File: cartography/cli.py [hunk 4]
```diff
@@ -1711,6 +1713,61 @@ def run(
...
+                int,
+                typer.Option(
+                    "--tenable-findings-lookback-days",
```
[the remark this answers is not in the document]
Comment (heryxpc) [reply cartography/cli.py:1761]: fixed

File: tests/integration/cartography/intel/tenable/test_findings.py [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+        "id",
+        "TenableFinding",
+        "id",
+        "RESOURCE",
+        rel_direction_right=True,
+    )
+    assert {
+        (TENABLE_TENANT_ID, FINDING_ID_1),
+        (TENABLE_TENANT_ID, FINDING_ID_2),
+    } <= actual_rels
```
[the remark this answers is not in the document]
Comment (heryxpc) [reply tests/integration/cartography/intel/tenable/test_findings.py:225]: fixed

File: cartography/models/tenable/plugins.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass
```
[the remark this answers is not in the document]
Comment (heryxpc) [reply cartography/models/tenable/plugins.py:1]: fixed

File: cartography/intel/tenable/api.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+from typing import Any
+
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+logger = logging.getLogger(__name__)
+
+_EXPORT_POLL_INTERVAL = 30  # seconds between status polls
+_MAX_POLL_ATTEMPTS = 60  # 30 minutes maximum wait
```
Comment (serge-wq) [inline cartography/intel/tenable/api.py:12]: Should the wait time also be configurable in case people want to wait more or less for the export to be available?

**Review by serge-wq — APPROVED** [review]

File: cartography/intel/tenable/api.py (hunk 7 above)
Comment (heryxpc) [reply cartography/intel/tenable/api.py:12]: Good callout. It is simple to configure the wait time/export poll interval. Tenable is [very cryptic on their rate limit periods](https://developer.tenable.com/docs/concurrency-limiting#concurrent-report-limits). During my tests it took less than 1 minute (1 call) to be ready.

I would like to keep it for 5 minutes, test it in produciton and add the configuration later with other changes I have, like exporting Web Application Scans.
